orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/TaskTasklet.groovy

```diff
@@ -143,19 +151,26 @@ class TaskTasklet implements Tasklet {
     return result
   }
 
-  private RepeatStatus cancel(Stage stage) {
+  private RepeatStatus cancel(Stage stage, boolean adjustStageStatusAndTasks = true) {
+    doCancel(stage, adjustStageStatusAndTasks)
+    return RepeatStatus.FINISHED
+  }
+
+  protected void doCancel(Stage stage, boolean adjustStageStatusAndTasks) {
```

Comment:
I'll write a test case around this cancel behavior as we don't currently have any coverage